parser/prism/Translator.cc

```diff
@@ -0,0 +1,2011 @@
+            return translateAssignment<pm_constant_path_write_node, void>(node);
+        }
+        case PM_CONSTANT_TARGET_NODE: { // Target of an indirect write to a constant
+            // ... like `TARGET1, TARGET2 = 1, 2`, `rescue => TARGET`, etc.
+            auto constantTargetNode = down_cast<pm_constant_target_node>(node);
+            return translateConst<pm_constant_target_node, parser::ConstLhs>(constantTargetNode);
+        }
+        case PM_CONSTANT_AND_WRITE_NODE: { // And-assignment to a constant, e.g. `C &&= false`
+            return translateOpAssignment<pm_constant_and_write_node, parser::AndAsgn, parser::ConstLhs>(node);
+        }
+        case PM_CONSTANT_OPERATOR_WRITE_NODE: { // Compound assignment to a constant, e.g. `C += 1`
+            return translateOpAssignment<pm_constant_operator_write_node, parser::OpAsgn, parser::ConstLhs>(node);
+        }
+        case PM_CONSTANT_OR_WRITE_NODE: { // Or-assignment to a constant, e.g. `C ||= true`
+            return translateOpAssignment<pm_constant_or_write_node, parser::OrAsgn, parser::ConstLhs>(node);
+        }
+        case PM_CONSTANT_READ_NODE: { // A single, unnested, non-fully qualified constant like `Foo`
+            auto constantReadNode = down_cast<pm_constant_read_node>(node);
+            return translateConst<pm_constant_read_node, parser::Const>(constantReadNode);
+        }
+        case PM_CONSTANT_WRITE_NODE: { // Regular assignment to a constant, e.g. `Foo = 1`
+            return translateAssignment<pm_constant_write_node, parser::ConstLhs>(node);
+        }
+        case PM_DEF_NODE: { // Method definitions, like `def m; ...; end` and `def m = 123`
+            auto defNode = down_cast<pm_def_node>(node);
+            auto declLoc = translateLoc(defNode->def_keyword_loc);
+            declLoc = declLoc.join(translateLoc(defNode->name_loc));
+
+            auto rparenLoc = defNode->rparen_loc;
+
+            if (rparenLoc.start != nullptr && rparenLoc.end != nullptr) {
+                declLoc = declLoc.join(translateLoc(defNode->rparen_loc));
+            }
+
+            auto name = parser.resolveConstant(defNode->name);
+
+            // These 2 need to be called on a new Translator with isInMethod set to true
+            Translator childContext = enterMethodDef();
+
+            unique_ptr<parser::Node> params;
+
+            // The definition has no parameters but still has parentheses, e.g. `def foo(); end`
+            // In this case, Sorbet's legacy parser will still hold an empty Args node, so we need to
+            // create one here
+            if (defNode->parameters == nullptr && rparenLoc.start != nullptr) {
+                params = make_unique<parser::Args>(location, NodeVec{});
+            } else {
+                params = childContext.translate(up_cast(defNode->parameters));
+            }
+
+            auto body = childContext.translate(defNode->body);
```

Comment:
The "2" here is a little disassociated from what it's referring to.  Instead, can this `childContext` be scoped to these two calls to make the intent clearer?

```c++
unique_ptr<parser::Node> params, body;

{
    Translator childContext = enterMethodDef();

    // translate params appropriately

    body = childContext.translate(defNode->body);
}
```

and then the comment is not necessary.

Reply:
Agreed that the comment isn't necessary.

There's actually a `translate` call below called on `this` instead of `childContext`. Once I fixed that, it's not really possible to collect the usages of the child context closer together, so now it spans the whole `case` block.